cloud/deployment/deployment.go

```diff
@@ -721,3 +728,39 @@ func GetDeploymentURL(deploymentID, workspaceID string) (string, error) {
 	}
 	return deploymentURL, nil
 }
+
+// printWarning lets the user know
+// when going from CE -> KE and if multiple queues exist,
+// a new default queue will get created.
+// If going from KE -> CE, it lets user know that a new default worker queue will be created.
+func printWarning(executor string, existingQLength int) {
+	if executor == KubeExecutor {
+		if existingQLength > 1 {
+			fmt.Println("\n Switching to KubernetesExecutor will replace all existing worker queues " +
+				"with one new default worker queue for this deployment.")
+		}
+	} else {
+		if executor == CeleryExecutor {
+			fmt.Println("\n Switching to CeleryExecutor will replace the existing worker queue" +
+				"with a new default worker queue for this deployment.")
+		}
+	}
+}
+
+// mutateExecutor updates currentSpec.Executor if requestedExecutor is not the same value.
+// It prints a helpful message describing the change and confirms the change with the user.
+// If a user does not confirm it, it returns false and does not make any changes to currentSpec.
+func mutateExecutor(requestedExecutor string, currentSpec astro.DeploymentCreateSpec, existingQLength int) (bool, astro.DeploymentCreateSpec) {
+	if requestedExecutor != "" && currentSpec.Executor != requestedExecutor {
+		// print helpful message describing the change
+		printWarning(requestedExecutor, existingQLength)
+		// confirm change with user
+		y, _ := input.Confirm("Are you sure you want to continue?")
```

Comment:
nit: There is already a confirmation before: https://github.com/astronomer/astro-cli/pull/1002/files#diff-b53f56e788c7abab81b9b7b016e1df0a87b7232006674c79a03bb7356c66c2ccR418-R426 might want to think about merging these two in single confirmation, also I think we should only ask for confirmation if `printWarning` is printing anything.

Reference:
```
neel@Neels-MacBook-Pro astro-cli % ./astro deployment update clcpyc462433332zzuikdax559 --executor=KubernetesExecutor 

Are you sure you want to update the neel-test-1 Deployment? (y/n) y
Are you sure you want to continue? (y/n) y
 NAME            NAMESPACE                  CLUSTER                   DEPLOYMENT ID                  RUNTIME VERSION                    DAG DEPLOY ENABLED     
 neel-test-1     asteroidal-zenith-0520     team-airflow-operator     clcpyc462433332zzuikdax559     7.1.0 (based on Airflow 2.5.0)     false                  

 Successfully updated Deployment
neel@Neels-MacBook-Pro astro-cli % 
```

Reply:
refactored the confirmation flow and consolidated all the destructive action confirmations in one spot.